### PR TITLE
Hide TestRunEnvironment params

### PR DESCRIPTION
**Pull request by sergioboi**

### CODE REVIEW

File: subprojects/android-test/test-inhouse-runner/src/main/kotlin/com/avito/android/runner/InHouseInstrumentationTestRunner.kt [hunk 1]
```diff
@@ -112,30 +113,24 @@ abstract class InHouseInstrumentationTestRunner :
...
-                            reportViewerUrl = reportConfig.reportViewerUrl,
-                            reportCoordinates = runEnvironment.testRunCoordinates,
-                            deviceName = DeviceName(runEnvironment.deviceName),
-                            logger = testReportLogger,
-                            reportsApi = ReportsApiFactory.create(
-                                host = reportConfig.reportApiUrl,
-                                loggerFactory = loggerFactory,
-                                httpClientProvider = httpClientProvider
-                            )
+        val transport: List<Transport> = when (val dest = runEnvironment.reportDestination) {
```
Comment (sergioboi) [inline subprojects/android-test/test-inhouse-runner/src/main/kotlin/com/avito/android/runner/InHouseInstrumentationTestRunner.kt:116]: Group parameters to `reportDestination` property

File: subprojects/android-test/test-inhouse-runner/src/main/kotlin/com/avito/android/runner/TestRunEnvironment.kt [hunk 2]
```diff
@@ -83,17 +72,30 @@ sealed class TestRunEnvironment {
...
         val outputDirectory: Lazy<File>,
-        val elasticConfig: ElasticConfig,
-        val sentryConfig: SentryConfig,
-        val statsDConfig: StatsDConfig,
-        val fileStorageUrl: String,
-        val reportConfig: ReportConfig?,
         val testRunCoordinates: ReportCoordinates,
-        val isSyntheticStepsEnabled: Boolean
-    ) : TestRunEnvironment()
+        internal val reportDestination: ReportDestination,
```
Comment (sergioboi) [inline subprojects/android-test/test-inhouse-runner/src/main/kotlin/com/avito/android/runner/TestRunEnvironment.kt:77]: Check Avito and mark internal

File: subprojects/android-test/test-inhouse-runner/src/main/kotlin/com/avito/android/runner/TestRunEnvironment.kt [hunk 3]
```diff
@@ -83,17 +72,30 @@ sealed class TestRunEnvironment {
...
+        internal val videoRecordingFeature: VideoFeatureValue,
+        internal val elasticConfig: ElasticConfig,
+        internal val sentryConfig: SentryConfig,
+        internal val statsDConfig: StatsDConfig,
+        internal val fileStorageUrl: String,
+    ) : TestRunEnvironment() {
+
+        sealed class ReportDestination {
+
+            class Backend(
```
Comment (sergioboi) [inline subprojects/android-test/test-inhouse-runner/src/main/kotlin/com/avito/android/runner/TestRunEnvironment.kt:88]: Group parameters

File: subprojects/android-test/test-inhouse-runner/src/main/kotlin/com/avito/android/runner/TestRunEnvironment.kt [hunk 4]
```diff
@@ -56,15 +52,8 @@ sealed class TestRunEnvironment {
...
 
-    data class RunEnvironment(
-        val isImitation: Boolean,
-
+    data class RunEnvironment internal constructor(
         val deviceId: String,
-        val deviceName: String,
-        val teamcityBuildId: Int,
-
-        val buildBranch: String,
```
Comment (sergioboi) [inline subprojects/android-test/test-inhouse-runner/src/main/kotlin/com/avito/android/runner/TestRunEnvironment.kt:66]: Unused

File: subprojects/android-test/test-inhouse-runner/src/main/kotlin/com/avito/android/runner/TestRunEnvironment.kt [hunk 5]
```diff
@@ -83,17 +72,30 @@ sealed class TestRunEnvironment {
          */
         val apiUrl: HttpUrl,
         val mockWebServerUrl: String,
-        val slackToken: String,
```
Comment (sergioboi) [inline subprojects/android-test/test-inhouse-runner/src/main/kotlin/com/avito/android/runner/TestRunEnvironment.kt:86]: Unused

File: subprojects/android-test/test-inhouse-runner/src/main/kotlin/com/avito/android/runner/TestRunEnvironment.kt [hunk 6]
```diff
@@ -145,22 +134,36 @@ fun provideEnvironment(
...
         )
     } catch (e: Throwable) {
         TestRunEnvironment.InitError(e.message ?: "Can't parse arguments for creating TestRunEnvironment")
     }
 }
 
+private fun parseReportDestination(argumentsProvider: ArgsProvider): ReportDestination {
+    val buildId = argumentsProvider.getMandatoryArgument("teamcityBuildId").toInt()
+    return if (buildId == TestRunEnvironment.LOCAL_STUDIO_RUN_ID) {
+        val isReportEnabled = argumentsProvider.getOptionalArgument("avito.report.enabled")?.toBoolean() ?: false
```
Comment (eugene-krivobokov) [inline subprojects/android-test/test-inhouse-runner/src/main/kotlin/com/avito/android/runner/TestRunEnvironment.kt:152]: I expect to disable any transport with `avito.report.enabled=false`, not only backend

**Review by eugene-krivobokov — APPROVED** [review]

File: subprojects/android-test/test-inhouse-runner/src/main/kotlin/com/avito/android/runner/TestRunEnvironment.kt (hunk 6 above)
Comment (sergioboi) [reply subprojects/android-test/test-inhouse-runner/src/main/kotlin/com/avito/android/runner/TestRunEnvironment.kt:152]: I agree with you, It's a problem with a property name
I don't want to change semantics in that PR

**Pull request merged**